lib/engine/bus_stops.ex

```diff
@@ -0,0 +1,105 @@
+defmodule Engine.BusStops do
+  alias Signs.Utilities.SignsConfig
+  use GenServer
+  require Logger
+
+  @bus_vehicle_type 3
+
+  @callback get_child_stop(String.t(), String.t(), String.t()) :: String.t()
+  def get_child_stop(parent_stop_id, route_id, direction_id) do
+    case :ets.lookup(:child_stops, {parent_stop_id, route_id, direction_id}) do
+      [{{^parent_stop_id, ^route_id, ^direction_id}, child_stop}] -> child_stop
+      _ -> nil
+    end
+  end
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, nil, name: __MODULE__)
+  end
+
+  def init(_) do
+    :ets.new(:child_stops, [:named_table, read_concurrency: true])
+    send(self(), :update)
+
+    {:ok,
+     %{
+       last_modified: nil,
+       all_bus_stop_ids: SignsConfig.all_bus_stop_ids()
+     }}
+  end
+
+  def handle_info(
+        :update,
+        %{last_modified: last_modified, all_bus_stop_ids: all_bus_stop_ids} = state
+      ) do
+    schedule_update(self())
+    api_url = Application.get_env(:realtime_signs, :api_v3_url)
+    api_key = Application.get_env(:realtime_signs, :api_v3_key)
+    http_client = Application.get_env(:realtime_signs, :http_client)
+    parent_route_directions = SignsConfig.all_bus_stop_route_direction_ids()
```

Comment:
Since we're doing repeated `in` lookups, we could convert this to a `MapSet` for a performance improvement.

Reply:
Ah good point. Thank you!